### PR TITLE
Implement select_item method

### LINKED ISSUE #484: select_item method
I need a `select_item` method defined as
```
select_item(x, t) == t.choose(x.T)
```

### PR DESCRIPTION
**Pull request by unnonouno**
fix #484 

I implemented simplest version, but we need to discuss about ideal specification.

### CODE REVIEW

File: chainer/functions/array/getitem.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        )
+
+    def forward_cpu(self, inputs):
+        x, t = inputs
+        return x[six.moves.range(t.size), t],
+
+    def forward_gpu(self, inputs):
+        x, t = inputs
+        y = cuda.elementwise(
+            'S t, raw T x, int32 n_channel',
```
Comment (delta2323) [inline chainer/functions/array/getitem.py]: Is there any reason why the order of `x` and `t` is reversed ?

File: chainer/functions/array/getitem.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            'raw T gx',
+            'gx[i * n_channel + t] = gloss',
+            'getitem_bwd'
+        )(t, gloss, x.shape[1], gx)
+        return gx, None
+
+
+def getitem(x, t):
+    """Get elements stored in given indicies.
+
```
Comment (delta2323) [inline chainer/functions/array/getitem.py]: I think it gets easier to understand if we add numpy code that explains the behavior like `getitem(x, t) = x[range(t.size), t]`

**Comment by delta2323** [comment]
> I implemented simplest version, but we need to discuss about ideal specification.

How about implement `cupy.choose`?

**Comment by delta2323** [comment]
This implementation fails when x and t is `cupy.ndarray` whose size are both 0. See #490 for details.

**Comment by unnonouno** [comment]
> How about implementing cupy.choose and make it backprop-able.
Humm... Its sounds good.

`__getitem__` method of `Variable` is another choice.

**Comment by unnonouno** [comment]
Yes, it looks like `numpy.choose`. But, I noticed that `getitem(x, a) == a.choose(x.T)`.

File: chainer/functions/array/select_item.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+class SelectItem(function.Function):
+
+    """Select elements stored in given indicies."""
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 2)
+
+        x_type, t_type = in_types
+        type_check.expect(
+            x_type.dtype.kind == 'f',
```
Comment (okuta) [inline chainer/functions/array/select_item.py]: This test is not need.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**